epoch.go

```diff
@@ -2854,72 +3026,72 @@ func (e *Epoch) processEmptyNotarization(emptyNotarization *EmptyNotarization) e
 	return e.processReplicationState()
 }
 
-func (e *Epoch) processLatestRoundReceived(latestRound *QuorumRound, from NodeID) error {
-	if latestRound == nil {
+// processQuorumRound processes a quorum round received from another node.
+// It verifies the quorum round and stores it in the replication state if valid.
+func (e *Epoch) processQuorumRound(round *QuorumRound, from NodeID) error {
+	if round == nil {
 		return nil
 	}
 
 	// make sure the latest round is well formed
-	if err := latestRound.IsWellFormed(); err != nil {
-		e.Logger.Debug("Received invalid latest round", zap.Error(err))
-		return err
+	if err := round.IsWellFormed(); err != nil {
+		return fmt.Errorf("received malformed latest round: %w", err)
 	}
 
-	if err := e.verifyQuorumRound(*latestRound, from); err != nil {
-		e.Logger.Debug("Received invalid latest round", zap.Error(err))
-		return err
+	if round.Finalization == nil && e.isVoteForFinalizedRound(round.GetRound()) {
+		return fmt.Errorf("received a quorum round for a too far behind. round: %d; seq: %d", round.GetRound(), round.GetSequence())
```

Comment:
"received a quorum round for a too far behind round".

But let's just state explicitly that the round has already been finalized.